Add assortativity tests

Relux-the-Relux opened the pull request:
There were no tests for assortativity so I added them.

angriman commented on networkit/cpp/correlation/test/AssortativityGTest.cpp:
```diff
@@ -0,0 +1,75 @@
+
+    Assortativity assort(G, attribute);
+    assort.run();
+    EXPECT_NEAR(coeff, assort.getCoefficient(), 1e-6);
```
In the previous test, you used a tolerance value of `1e-8`, why are we using a different value here?

angriman reviewed (commented):
Thank you for the new tests. It would be helpful a brief explanation of how the ground truth was computed (the two `coeff` values).

Relux-the-Relux commented:
> Thank you for the new tests. It would be helpful a brief explanation of how the ground truth was computed (the two `coeff` values).

Added explanation where the truth value comes from and fixed a small bug that didn't properly consider undirected graphs in the partition variant.

angriman commented on networkit/cpp/correlation/Assortativity.cpp:
```diff
@@ -36,6 +36,7 @@ void Assortativity::run() {
         std::vector<std::vector<double>> E(k, std::vector<double>(k, 0.0));
         G->forEdges([&](node u, node v) {
             E[P[u]][P[v]] += 1;
+            if(!G->isDirected() && P[u] != P[v]) E[P[v]][P[u]] += 1;
```
```suggestion
            if(!G->isDirected() && P[u] != P[v])
                E[P[v]][P[u]] += 1;
```

angriman commented on networkit/cpp/correlation/test/AssortativityGTest.cpp:
```diff
@@ -0,0 +1,91 @@
+    G.addEdge(6, 8);
+
+    /*
+    Since we have 2 edges between each partition(except partion 0 and 2) and 2 edges within each.
```
```suggestion
    Since we have 2 edges between each partition (except partions 0 and 2) and 2 edges within each.
```

angriman commented on networkit/cpp/correlation/test/AssortativityGTest.cpp:
```diff
@@ -0,0 +1,91 @@
+
+    /*
+    Since we have 2 edges between each partition(except partion 0 and 2) and 2 edges within each.
+    This gives us the matrix(normalized by m):
```
```suggestion
    This gives us the matrix (normalized by m):
```

angriman commented on networkit/cpp/correlation/test/AssortativityGTest.cpp:
```diff
@@ -0,0 +1,91 @@
+    |0.2 0.2 0.2
+    |0   0.2 0.2
+
+    The diagSum is then 0.6 and the ab coeff sum is 0.68
```
```suggestion
    The diagSum is thus 0.6 and the ab coeff sum is 0.68
```
What exactly is the "ab coeff"?
